test/extensions/filters/http/admission_control/admission_control_filter_test.cc

```diff
@@ -0,0 +1,249 @@
+                                        std::move(tls), controller_, evaluator_);
+  }
+
+  void setupFilter(std::shared_ptr<AdmissionControlFilterConfig> config) {
+    filter_ = std::make_shared<AdmissionControlFilter>(config, "test_prefix.");
+    filter_->setDecoderFilterCallbacks(decoder_callbacks_);
+  }
+
+  void sampleGrpcRequest(std::string&& grpc_status) {
+    Http::TestResponseHeaderMapImpl headers{{"content-type", "application/grpc"},
+                                            {"grpc-status", grpc_status}};
+    filter_->encodeHeaders(headers, true);
+  }
+
+  void sampleHttpRequest(std::string&& http_error_code) {
+    Http::TestResponseHeaderMapImpl headers{{":status", http_error_code}};
+    filter_->encodeHeaders(headers, true);
+  }
+
+protected:
+  std::string stats_prefix_{""};
+  NiceMock<Runtime::MockLoader> runtime_;
+  NiceMock<Server::Configuration::MockFactoryContext> context_;
+  Stats::IsolatedStoreImpl scope_;
+  Event::SimulatedTimeSystem time_system_;
+  NiceMock<Runtime::MockRandomGenerator> random_;
+  std::shared_ptr<AdmissionControlFilter> filter_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks_;
+  NiceMock<MockThreadLocalController> controller_;
+  std::shared_ptr<MockResponseEvaluator> evaluator_;
+  const std::string default_yaml_{R"EOF(
+enabled:
+  default_value: true
+  runtime_key: "foo.enabled"
+sampling_window: 10s
+aggression_coefficient:
+  default_value: 1.0
+  runtime_key: "foo.aggression"
+success_criteria:
+  http_criteria:
+  grpc_criteria:
+)EOF"};
+};
+
+// Ensure the filter can be disabled/enabled via runtime.
+TEST_F(AdmissionControlTest, FilterRuntimeOverride) {
+  const std::string yaml = R"EOF(
+enabled:
+  default_value: true
+  runtime_key: "foo.enabled"
+sampling_window: 10s
+aggression_coefficient:
+  default_value: 1.0
+  runtime_key: "foo.aggression"
+success_criteria:
+  http_criteria:
+  grpc_criteria:
+)EOF";
+
+  auto config = makeConfig(yaml);
+  setupFilter(config);
+
+  // "Disable" the filter via runtime.
+  EXPECT_CALL(runtime_.snapshot_, getBoolean("foo.enabled", true)).WillRepeatedly(Return(false));
+
+  // The filter is bypassed via runtime.
+  EXPECT_CALL(controller_, requestTotalCount()).Times(0);
+  EXPECT_CALL(controller_, requestSuccessCount()).Times(0);
+
+  // We expect no rejections.
+  Http::RequestHeaderMapImpl request_headers;
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
+}
+
+// Ensure the filter disregards healthcheck traffic.
+TEST_F(AdmissionControlTest, DisregardHealthChecks) {
+  auto config = makeConfig(default_yaml_);
+  setupFilter(config);
+
+  StreamInfo::MockStreamInfo stream_info;
+  EXPECT_CALL(decoder_callbacks_, streamInfo()).WillOnce(testing::ReturnRef(stream_info));
+  EXPECT_CALL(stream_info, healthCheck()).WillOnce(Return(true));
+
+  // Fail lots of requests so that we would normally expect a ~100% rejection rate. It should pass
```

Comment:
i only see one request being issued here? is this comment up to date?

Reply:
It sure isn't up to date. Thanks.